components/datetime/src/skeleton/helpers.rs

```diff
@@ -232,8 +232,7 @@ pub fn create_best_pattern_for_fields<'data>(
             date_patterns.for_each_mut(|pattern| {
                 let date = pattern.clone();
                 let time = time_pattern.clone();
-                #[allow(clippy::expect_used)]
-                // TODO(#1668) Clippy exceptions need docs or fixing.
+                #[allow(clippy::expect_used)] // Generic pattern combination should never fail.
```

Comment:
Question: Why is the API fallible, then? What preconditions of the API are always satisfied, and can you write those down, e.g. in debug assertions?

Reply:
`combined` is a fallible constructor because I designed it to gracefully handle scenarios of incorrect patterns. But the way `skeletons` are written assumes that patterns are correct.

If you want to propagate error scenario, we'd have to turn `create_best_pattern_for_fields` to return Result and likely all its callers. @gregtatum was the original author of that code and I assume he did a shortcut because of how we treat skeletons now.

The only precondition we assume here is that the pattern comes from CLDR and is valid (currently `combined` will error out when encountering a pattern with a placeholder different than `{0}` or `{1}`).

debug assert would just iterate over the pattern and verify that all placeholders are only 0 and 1. I'm not sure how much more useful it is over `expect` here especially that in theory `combined` could change later and add different errors, and we'd then have to revisit this debug assert to cover them too.

I'm against changes here at this moment.

Reply:
So, what you're basically saying is this code doesn't conform to the ICU4X style guide.

Okay for now, but only because this is experimental code that is likely to be changed. The objective of this ticket is to clean up areas where we assume things about data that we shouldn't be assuming; in some cases, that means refactoring or rewriting code that was written before we enforced this style in the project.

Please replace this TODO with a new one pointing to a new ticket that says we still have work to do in the datetime crate.

Reply:
Filed #2626